crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -165,80 +174,298 @@ declare_source_rule! {
 
 impl Rule for OrganizeImports {
     type Query = Ast<JsModule>;
-    type State = State;
+    type State = Box<[Issue]>;
     type Signals = Option<Self::State>;
     type Options = Options;
 
-    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        struct ChunkBuilder {
+            slot_indexes: std::ops::Range<u32>,
+            max_key: ImportKey,
+        }
+        impl ChunkBuilder {
```

Comment:
Could you add some newlines between blocks/segments? This is an almost 100 line function without any newlines for readability. It's kinda overwhelming.